lib/mix/tasks/ash.gen.resource.ex

```diff
@@ -0,0 +1,230 @@
+      define :update
+      define :destroy
+    end
+    """
+
+    data_layer_content =
+      if uses_ash_postgres?() do
+        """
+        use Ash.Resource,
+          data_layer: AshPostgres.DataLayer
+
+        postgres do
+          table "#{table_name}"
+          repo #{app_name}.Repo
+        end
+        """
+      else
+        """
+        use Ash.Resource, data_layer: Ash.DataLayer.Ets
+        """
+      end
+
+    resource_file_content = """
+    defmodule #{resource_module_name} do
+      #{data_layer_content}
+
+      attributes do
+        uuid_primary_key :id
+        #{attribute_definitions}
+      end
+
+      actions do
+        defaults [:create, :read, :update, :destroy]
+      end
+
+      #{unless no_code_interface?, do: code_interface_content}
+    end
+    """
+
+    api_dir_path = "lib/#{String.replace(api_module_name, ".", "/") |> String.downcase()}"
+    resource_dir_path = "#{api_dir_path}/resources"
+
+    # Ensure directories exist
+    File.mkdir_p!(api_dir_path)
+    File.mkdir_p!(resource_dir_path)
+
+    api_file_path = "#{api_dir_path}.ex"
+    resource_file_path = "#{resource_dir_path}/#{String.downcase(resource_name)}.ex"
+
+    if File.exists?(api_file_path) do
+      content = File.read!(api_file_path)
+
+      unless String.contains?(content, "resource #{resource_module_name}") do
+        # Append the new resource to the existing API file
+        parts = String.split(content, "\n")
+        index = Enum.find_index(parts, fn line -> String.trim(line) == "end" end)
+
+        if index do
+          parts = List.insert_at(parts, index, "    resource #{resource_module_name}")
+          File.write!(api_file_path, Enum.join(parts, "\n"))
+          IO.puts("Modified #{api_file_path}")
+        end
+      end
+    else
+      File.write!(api_file_path, """
+      defmodule #{api_module_name} do
+        use Ash.Api
+
+        resources do
+          resource #{resource_module_name}
+        end
+      end
+      """)
+
+      IO.puts("Generated #{api_file_path}")
+    end
+
+    # Check if the API is listed in the configuration of the current application
+    unless api_listed?(api_module_name) do
+      IO.puts("""
+      \e[31m
+      WARNING:
+      #{api_module_name} is not yet listed in the application's configuration.\e[0m
+      Make sure to add it to the configuration. Here is an example:
+
+      ** config/config.exs **
+      import Config
+
+      config :#{Macro.underscore(app_name)}, :ash_apis, [#{api_module_name}]
+
+      """)
+    end
+
+    if File.exists?(resource_file_path) do
+      IO.puts("File #{resource_file_path} already exists. Replace? [Yn]")
+      response = IO.gets("> ") |> String.trim()
+
+      if String.downcase(response) not in ["y", "yes", ""] do
+        IO.puts("Aborted!")
+        exit(:normal)
+      end
+    end
+
+    File.write!(resource_file_path, resource_file_content)
+    IO.puts("Generated #{resource_file_path}")
+
+    if uses_ash_postgres?() do
```

Comment:
Honestly, I'm still pretty torn about putting `ash_postgres` specific code generation in Ash core. The idea that we're working towards behind the scenes in terms of generators is that we would compose generator modifiers from various extensions that would be able to add their own snippets. i.e `--with postgres,graphql` or something like that. While having `postgres` configured automatically would be great I'd rather, at least to start, generate the resource without `ash_postgres` related code for now. We can always add it back in in a follow up.